spring-cloud-kubernetes-examples/kubernetes-leader-election-example/README.md

```diff
@@ -0,0 +1,83 @@
+```
+
+## Overview
+
+Spring Cloud Kubernetes leader election mechanism implements leader election API of Spring Integration using Kubernetes ConfigMap.
+Multiple instances of the same application compete for a leadership of a specified role.
+But only one of them can become a leader and receive an `OnGrantedEvent` application event with a leadership `Context`.
+All the instances periodically try to get a leadership and whichever comes first - becomes a leader.
+The new leader will remain until either its instance disappears from the cluster or it yields its leadership.
+Once the leader is gone, any of the existing instances can become a new leader (even the previous leader if it yielded leadership but stayed in the cluster). 
+And finally, if the leadership is yielded or revoked for some reason, the old leader receives `OnRevokedEvent` application event.
+
+## Example application usage
+
+To begin with, build and deploy the application:
+```
+mvn clean package fabric8:deploy -Pkubernetes
+```
+
+This will deploy a single application instance to the cluster and that instance will automatically become a leader.
+
+Create an environment variable for an easier application access:
+```
+SERVICE_URL=$(minikube service kubernetes-leader-election-example --url)
+```
+
+Get leadership information:
+```
+curl $SERVICE_URL
+```
+
+You should receive a message like this:
+```
+I am 'kubernetes-leader-election-example-1234567890-abcde' and I am the leader of the 'world'
+```
+
+Yield the leadership:
+```
+curl -X PUT $SERVICE_URL
+```
+
+And check the leadership information again:
+```
+curl $SERVICE_URL
+```
+
+Now you should receive a message like this:
+```
+I am 'kubernetes-leader-election-example-1234567890-abcde' but I am not a leader of the 'world'
+```
+
+If you wouldn't do anything for a few seconds, the same instance will become a leader again because it only yielded its leadership but stayed in the cluster.
+
+Now scale the application to two instances and try all the steps again:
+```
+kubectl scale --replicas=2 deployment.apps/kubernetes-leader-election-example
+```
+
+> Note: with multiple replicas in the cluster, `curl` command will access one of them depending on the Kubernetes load balancing configuration.
+Thus, when trying to yield the leadership, request might go to a non-leader node first. Just execute command again until it reaches the correct node.
+
+> Note: instances periodically try to acquire leadership and Spring Cloud Kubernetes doesn't decide which one of them is more worth to become one.
+Thus, it is possible that the instance which just yielded the leadership, made another leadership take over request faster than another instances and became a leader again.
+
+## Access control notice 
+
+Leader election mechanism uses Kubernetes ConfigMap feature to coordinate leadership information.
+In order to access it, [Role](./src/main/fabric8/role.yaml) and [RoleBinding](./src/main/fabric8/rb.yaml) objects are defined. 
```

Comment:
unless i miss something, in most cases the defaults should cover us without having to modify roles etc. do we really need to include rbac related resources?

Reply:
Without this ConfigMap cannot be viewed